politeiawww/user.go

```diff
@@ -1656,47 +1656,101 @@ func (p *politeiawww) processManageUser(mu *www.ManageUser, adminUser *user.User
 	return &www.ManageUserReply{}, nil
 }
 
-// processUsers returns a list of users given a set of filters.
-func (p *politeiawww) processUsers(users *www.Users) (*www.UsersReply, error) {
+// processUsers returns a list of users given a set of filters. Admins
+// can search by pubkey, username or admin (partial matches returned).
+// Non admins can search by pubkey or username (only exact matches returned).
+func (p *politeiawww) processUsers(users *www.Users, isAdmin bool) (*www.UsersReply, error) {
 	var reply www.UsersReply
-	reply.Users = make([]www.AbridgedUser, 0)
+	reply.Users = make([]www.AbridgedUser, 0, www.UserListPageSize)
+	var u *user.User
+	var userID string
 
 	emailQuery := strings.ToLower(users.Email)
 	usernameQuery := formatUsername(users.Username)
+	pubkeyQuery := users.PublicKey
 
-	err := p.db.AllUsers(func(user *user.User) {
-		reply.TotalUsers++
-		userMatches := true
+	// Get user by pubkey, if provided
+	if pubkeyQuery != "" {
+		// Ensure we got a proper pubkey, if provided
+		_, err := validatePubkey(pubkeyQuery)
+		if err != nil {
+			return nil, err
+		}
 
-		// If both emailQuery and usernameQuery are non-empty, the user
-		// must match both to be included in the results.
-		if emailQuery != "" {
-			if !strings.Contains(strings.ToLower(user.Email),
-				emailQuery) {
-				userMatches = false
-			}
+		// Get userID by pubkey
+		id, ok := p.userPubkeys[pubkeyQuery]
+		if !ok {
+			return nil, fmt.Errorf("pubkey not found")
```

Comment:
I agree. Have removed error checking. I was actually trying to copy what `getUserIDByPubKey()` is doing. Which returns the `ok`. However, in this instance, if `p.userPubkeys[pubkeyQuery]` does not return a match, it will (according to the golang spec) return the zero value for the map type (a blank string ``). If this happens, the getUserByIDStr() will catch it and return an error. So it appears OK to remove the ok.

Reply:
This still isn't right.  If I send a valid pubkey that doesn't correspond to any user, I'm getting a `400, invalid user UUID`.  I would expect it to return a successful response with no matches found.